src/mlpack/core/tree/binary_space_tree/midpoint_split_impl.hpp

```diff
@@ -65,18 +69,20 @@ bool MidpointSplit<BoundType, MatType>::SplitNode(const BoundType& bound,
       {
         maxWidth = width;
         splitDimension = d;
+
+        // Split in the midpoint of that dimension.
+        splitVal = ranges[d].Mid();
       }
     }
 
     delete[] ranges;
   }
 
+  assert(splitVal != DBL_MAX);
+
   if (maxWidth == 0) // All these points are the same.  We can't split.
     return false;
```

Comment:
Hm, I think we could avoid the assert here (or handle it better) if we just made the conditional `if (maxWidth <= 0)`; what do you think?  Then no split would be performed.